### DISCUSSION TITLE
Select Filter query doesn't work option 'Show All'

### DISCUSSION THREAD

**Discussion by Ivcast20** [rappasoft/laravel-livewire-tables#1013] [Q&A] [answered]
I'm doing a datatable with products table and this table has a foreign key to categories, but when I make the filter I doesn't show me the All option
this is my code
![image](https://user-images.githubusercontent.com/59401990/211886431-a11a923b-03f2-4047-94d1-d845c21e992f.png)

I tried to add show all in the options
![image](https://user-images.githubusercontent.com/59401990/211886645-ef932b89-74fc-4f26-bf30-f27da9ee97be.png)

but I got this error
![image](https://user-images.githubusercontent.com/59401990/211886760-90c27c4b-096a-4ffc-8db7-6979dc933619.png)

**Comment by Ivcast20** [comment, marked as answer]
I think i have the answer, but i don't know i there's other way of doing it.
All I had to do was this.
![image](https://user-images.githubusercontent.com/59401990/211889501-a13a5858-a9cd-4728-84af-0749e760daa9.png)

**Reply by Ivcast20** [reply]
Please, If something is wrong, tell me.

**Comment by lrljoe** [comment]
@Ivcast20 - no, you've done that correctly!

I have however opened a Pull Request to add in a "setFirstOption()" function for the Select Filter.  Not sure if/when @rappasoft will merge this, I've been quite active in creating pull requests lately, so he's got quite a few to look at!
You are of course welcome to extend the SelectFilter class, and implement the functionality now (or amend the vendor files - at your own risk), should you so wish!

[Pull Request](https://github.com/rappasoft/laravel-livewire-tables/pull/1016)
Alternatively, you could temporarily switch to my repo, which has this and another number of newer pull requests.
[View Repo](https://github.com/rappasoft/laravel-livewire-tables/discussions/1018)
